### PR TITLE
new resource `azurerm_spring_cloud_container_deployment`

### PR DESCRIPTION
**Pull request by ms-henglu**

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_container_deployment_resource.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+				Required:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"args": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+				},
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_container_deployment_resource.go:70]: we should validate this?

File: internal/services/springcloud/spring_cloud_container_deployment_resource.go [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+			},
+
+			"commands": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				Elem: &pluginsdk.Schema{
+					Type: pluginsdk.TypeString,
+				},
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_container_deployment_resource.go:78]: and this?

File: internal/services/springcloud/spring_cloud_container_deployment_resource.go [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+				Type:         pluginsdk.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 500),
+			},
+
+			"language_framework": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
```
Comment (katbyte) [inline internal/services/springcloud/spring_cloud_container_deployment_resource.go:99]: we can probably validate this better?

File: website/docs/r/spring_cloud_container_deployment.html.markdown [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+* `args` - (Optional) Arguments to the entrypoint. The docker image's CMD is used if this is not provided.
+
+* `commands` - (Optional) Entrypoint array. Not executed within a shell. The docker image's ENTRYPOINT is used if this is not provided.
+
+* `environment_variables` - (Optional) Specifies the environment variables of the Spring Cloud Deployment as a map of key-value pairs.
+
+* `instance_count` - (Optional) Specifies the required instance count of the Spring Cloud Deployment. Possible Values are between `1` and `500`. Defaults to `1` if not specified.
+
+* `language_framework` - (Optional) Language framework of the container image uploaded.
```
Comment (katbyte) [inline website/docs/r/spring_cloud_container_deployment.html.markdown:76]: grammer

File: website/docs/r/spring_cloud_container_deployment.html.markdown [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+* `spring_cloud_app_id` - (Required) The ID of the Spring Cloud Service. Changing this forces a new Spring Cloud Container Deployment to be created.
+
+* `image` - (Required) Container image of the custom container. This should be in the form of <repository>:<tag> without the server name of the registry.
+
+* `server` - (Required) The name of the registry that contains the container image.
+
+---
+
+* `args` - (Optional) Arguments to the entrypoint. The docker image's CMD is used if this is not provided.
```
Comment (katbyte) [inline website/docs/r/spring_cloud_container_deployment.html.markdown:68]: grammer

File: website/docs/r/spring_cloud_container_deployment.html.markdown [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+* `commands` - (Optional) Entrypoint array. Not executed within a shell. The docker image's ENTRYPOINT is used if this is not provided.
```
Comment (katbyte) [inline website/docs/r/spring_cloud_container_deployment.html.markdown:70]: grammer

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ms-henglu - left some comments inline

File: website/docs/r/spring_cloud_container_deployment.html.markdown [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Spring Cloud Container Deployment. Changing this forces a new Spring Cloud Container Deployment to be created.
+
+* `spring_cloud_app_id` - (Required) The ID of the Spring Cloud Service. Changing this forces a new Spring Cloud Container Deployment to be created.
+
+* `image` - (Required) Container image of the custom container. This should be in the form of <repository>:<tag> without the server name of the registry.
```
[the remark this answers is not in the document]
Comment (ms-henglu) [reply website/docs/r/spring_cloud_container_deployment.html.markdown:62]: Maybe not necessary? It seems https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/container_group_resource.go#L191 has same schema.

**Comment by ms-henglu** [comment]
I've added a commit to address the issues, please take another look, thanks! @katbyte

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 🚜

**Pull request merged**